xpdan/db_utils.py

```diff
@@ -0,0 +1,100 @@
+from pprint import pprint
+from .utils import _timestampstr
+
+
+def need_name_here(hdrs, key, verbose=True):
+    """Build a dictionary where the keys are the unique values for the given
```

Comment:
In a list of hdrs, group the scans by header-key.

Use this function to find all the scans in the list of headers that have a particular key value, such as 'sample_name'='Ni'

function returns a list of indices for the position in the list, associated with the metadata key values.